browser/ephemeral_storage/ephemeral_storage_tab_helper.h

```diff
@@ -43,6 +44,7 @@ class EphemeralStorageTabHelper
   friend class content::WebContentsUserData<EphemeralStorageTabHelper>;
   scoped_refptr<content::SessionStorageNamespace> local_storage_namespace_;
   scoped_refptr<content::SessionStorageNamespace> session_storage_namespace_;
+  scoped_refptr<content::TLDEphemeralStorage> tld_ephemeral_storage_;
```

Comment:
why is this a scoped_refptr?

Reply:
Reference-counting is used to manage the lifetime of all ephemeral storage. This matches how the lifetime of storage is matched for DOM storage via `SessionStorageNamespace` in upstream Chromium and in our ephemeral storage implementation.

Reply:
ok, I see what's happening now with the weakptr map